### PR TITLE
Add 2019-05-07 Callum DevOps Talk

### PR DESCRIPTION
**Pull request by MichAnnTan**

### CODE REVIEW

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+**2. Ease of operations.**
+
+"We’re going to run it as a service so we care about operations," said Styan. The team was determined to incorporate the best of both [microservices and monoliths](https://grafana.com/blog/2019/04/15/how-we-designed-loki-to-work-easily-both-as-microservices-and-as-monoliths/), whether you run a single process on bare metal or microservices at hyperscale. 
+
+"Internally we run Loki, based on Cortex, as a set of microservices," said Styan. "These microservices make the system multi-tenant and horizontally scalable. For example, there’s services for sharding and replicating data, for ingestion, querying, etc."
+
+Reminder, however: Loki was designed to be a tool for incident investigations. "We’re not trying to replace things like Splunk or ELK if you’re doing analytics or business intelligence purposes," said Styan. "Please don’t parse structured logs to generate time series data."
```
Comment (cstyan) [inline content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: this isn't really relevant to this "ease of operations" section, should be later in the description of what loki is/how it works I think

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+Here we have a basic deployment diagram. The blue line in the middle is Promtail talking to Kubernetes service discovery. The red lines are Promtail collecting from logs streams, in this case pods, and the green line is Promtail sending those logs to Loki. 
+SLIDE 
+
+Here we have a Kubernetes cluster running some services app1 and app2. The blue line in the middle is Promtail using k8s SD to ask about what pods or services are running. The red lines on the right are Promtail collecting from the log streams it has found via k8s SD. The green line is Promtail sending logs to Loki, and the blue line is Grafana running queries against the Loki data source.
+SLIDE 
+
+Learn more about Loki on [Github]( https://github.com/grafana/loki) and check out its interface [here](https://grafana.com/blog/2019/01/02/closer-look-at-grafanas-user-interface-for-loki/).
```
Comment (cstyan) [inline content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: we could link to davids talk as well: https://www.youtube.com/watch?v=U7C5SpRtK74

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money (hunk 1 above)
Comment (Kuqd) [inline content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: > said Styan. "Please don’t parse structured logs to generate time series data."

We actually plan to parse log lines at least from promtail to extract new labels that will be indexed. The point is still correct but I think we should may be reword it something like:

> Please don't generate time series data from log queries.

WDYT

Comment (cstyan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: How about?

> While we have plans to have Loki parse structured logs to find more labels, we don't recommend generating time series data from log queries with any tool.

**Comment by cstyan** [comment]
should we mention that this is a summary of the devopsdays talk?

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+cover_opacity = "0.4"
+cover_blur = "1px"
+description = "A look at how Loki creates a streamlined UI to correlate metrics and logs."
+excerpt = "Instead of grepping through logs, Grafana Labs Software Engineer Callum Styan explains how Loki makes log aggregrgation for incident investigation easier."
+categories = ["Grafana", "Loki"]
++++
+
+##
+
+The situation is all too familiar: You get an alert. You look at your metrics and your dashboards to try and find out what the cause might be when the incident actually started (instead of when the alert happened). Then you have to go somewhere else to look at logs because eventually you need more data.
```
[the remark this answers is not in the document]
Comment (MichAnnTan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: Done

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+##
+
+The situation is all too familiar: You get an alert. You look at your metrics and your dashboards to try and find out what the cause might be when the incident actually started (instead of when the alert happened). Then you have to go somewhere else to look at logs because eventually you need more data. 
+
+"The main problem with this is that you don’t have one single UI," explained Grafana Labs Software Engineer Callum Styan at [Devopsdays](https://www.devopsdays.org/) in [Vancouver](https://www.devopsdays.org/events/2019-vancouver/welcome/). "You have to jump all over the place to find the data you actually want." 
+
+To add more layers to an already complicated process, there are likely several cross-functional teams who manage data in different ways. "In most orgs, there are probably multiple tools for the same thing. One team might use Splunk while another employs ElasticSearch," said Styan. 
+
+And when it comes to accessing logs in Kubernetes, "some of us are infrastructure engineers but your application engineers might not know how to do that," said Styan. "What if they want to grep, but they don’t know where to go to get the logs?"
```
[the remark this answers is not in the document]
Comment (MichAnnTan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: Done

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+INSERT TWEET IMAGE HERE 
+
+With existing log aggregation solutions, "it’s a pain to even find the logs for the service that I care about," said Styan. "I would really rather dump the log to standard out and grep through it."
```
[the remark this answers is not in the document]
Comment (MichAnnTan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: Done

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money (hunk 1 above)
Comment (MichAnnTan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: Moved but pls review to make sure it is where you think appropriate

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money (hunk 2 above)
Comment (MichAnnTan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: Done

File: content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money (hunk 1 above)
Comment (MichAnnTan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: Done

Comment (Kuqd) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: yes that sounds perfect.

Comment (cstyan) [reply content/blog/2019-05-07-how-loki-correlates-metrics-logs-saves-money]: seems like it's in the same place still?

we could also mention that maintaining the inverted index of all log contents is one of the biggest pain points when it comes to operating existing log aggregation tools (and a factor in the cost of SaaS options) in addition to the indexes storage cost

**Pull request merged**